datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -499,6 +492,7 @@ pub fn coercion_decimal_mathematics_type(
     left_decimal_type: &DataType,
     right_decimal_type: &DataType,
 ) -> Option<DataType> {
+    // TODO: Move this logic into kernel implementations
```

Comment:
I intend to this as a follow on, it is peculiar that BinaryExpr is performing a precision/scale coercion prior to calling kernels, which then in the case of divide then do their own scale coercion. The kernels should just be smart enough to perform the arithmetic correctly

Reply:
FYI @viirya

Reply:
Sounds good to me, if the coercion rule that actually comes from Spark is considered to be general enough for all downstreams.